sleap_io/model/instance.py

```diff
@@ -0,0 +1,312 @@
+from __future__ import annotations
+from attrs import define, validators
+import attr
+from typing import List, Optional, Tuple, Union, Dict
+from sleap_io.model.video import Video
+from sleap_io.model.skeleton import Skeleton, Node
+import numpy as np
+
+
+@define(auto_attribs=True)
+class Point:
+    """A labeled point and any metadata associated with it.
+
+    Args:
+        x: The horizontal pixel location of point within image frame.
+        y: The vertical pixel location of point within image frame.
+        visible: Whether point is visible in the labelled image or not.
+        complete: Has the point been verified by the user labeler.
+    """
+
+    x: float = np.nan
+    y: float = np.nan
+    visible: bool = attr.ib(default=True, kw_only=True)
+    complete: bool = attr.ib(default=False, kw_only=True)
+
+
+@define(auto_attribs=True)
+class PredictedPoint(Point):
+    """A predicted point is an output of the inference procedure.
+
+    It has all the properties of a labeled point, plus a score.
+
+    Args:
+        score: The point-level prediction score.
+    """
+
+    score: float = 0.0
+
+    @classmethod
+    def from_point(cls, point: Point, score: float = 0.0) -> PredictedPoint:
+        """Create a PredictedPoint from a Point
+
+        Args:
+            point: The point to copy all data from.
+            score: The score for this predicted point.
+
+        Returns:
+            A scored point based on the point passed in.
+        """
+        return cls(
+            x=point.x,
+            y=point.y,
+            visible=point.visible,
+            complete=point.complete,
+            score=score,
+        )
+
+
+# "By default, two instances of attrs classes are equal if all their fields are equal."
+@define(auto_attribs=True, eq=True)
+class Track:
+    """A track object is associated with a set of animal/object instances
```

Comment:
instances -> `Instance` s (with the backticks and a space between the backtick and last "s")